Start basic Git styleguide

duggiefresh opened the pull request:
This is not intended to be the full encompassing guide on Git, and only
a basic guide. I'd like to provide basic suggestions to the team, in order
to make our commits more consistent. Project teams and individuals can use
this as a guide, and are free to make modifications as they see fit.

Signed-off-by: Doug Yun <doug.yun@dockyard.com>

aadam21 commented on general/git.md:
````diff
@@ -0,0 +1,157 @@
+```bash
+Capitalized, short (50 chars or less) summary
+
+More detailed explanatory text, if necessary.  Wrap it to about 72
````
Yes!

aadam21 commented on general/git.md:
```diff
@@ -0,0 +1,157 @@
+
+How do we inform our audience?
+
+Let's take a look at some examples of commits for an Emac's init file.
```
Wouldn't this be `Emacs`?

aadam21 commented on general/git.md:
```diff
@@ -0,0 +1,157 @@
+
+# Ideal
+
+$ git log --oneline -1
```
Do you think there would be any value lost in removing this and illustrating the "happy path" with the `--grep` used below to eliminate some duplication?

duggiefresh replied:
Good point! I only included the `git log` portion to show that there was a commit title that was descriptive with important keywords. If you find it to be duplicative work, I can remove the `git log` and highlight that the descriptive title made the `grep` possible.

aadam21 replied:
I don't honestly have a strong opinion either way. I just noticed on first reading I had to do a double take to make sure I didn't re-read or misread a line.

monicawheeler commented on general/git.md:
```diff
@@ -0,0 +1,157 @@
+Commit messages are an opportunity for us to convey a brief story or synopsis of
+work completed. We should be answering questions like:
+
+*What caused this original regression and why does this newly introduced code fix it?*,
```
_What do you think of changing the format, and some minor grammar updates, to this?_

Commit messages are an opportunity for us to convey a brief story or synopsis of
work completed. We should be answering questions like:

- What caused this original regression and why does this newly introduced code fix it?
- What dependency was removed from the application, and how does this affect performance?

When commit messages answer these questions, our audience, colleagues and clients, will have a better understanding of what our intention was.

monicawheeler commented on general/git.md:
```diff
@@ -0,0 +1,157 @@
+When commit messages answer questions like these, our audience - colleagues and clients -
+will have a better understanding in time of what our intention was.
+
+* Descriptive commit title
```
_It might be helpful to add what these bulleted items represent. For example:_

The following is a list of items you should include in your commit message: 

* Descriptive commit title
...
...

monicawheeler commented on general/git.md:
```diff
@@ -0,0 +1,157 @@
+
+### Writing Message Titles
+
+Commit message titles should be descriptive and succinct. Imagine that you
```
Recommended change:

```
Commit message titles should be descriptive and succinct. Imagine that you
have to scan a list of commit titles and messages. What characteristics would
make that task easier?
```

monicawheeler commented on general/git.md:
````diff
@@ -0,0 +1,157 @@
+
+```bash
+
+# Not ideal
````
Love having the examples showing ideal and not ideal solutions! What do you think about pulling the `# Not ideal` and `# Ideal` out of the code snippets and into a subheader to clearly label the types of code snippets?

duggiefresh replied:
👍 I've pulled out the "not ideal" and "ideal" into subheaders. I also went with "not recommended" and "recommended." Any thoughts on the word update?

monicawheeler commented on general/git.md:
```diff
@@ -0,0 +1,157 @@
+
+### Writing Message Bodies
+
+Just like descriptive commit titles, the content of the commit can provide
```
💯 Love this!

duggiefresh commented:
Thanks all for commenting! I've updated the PR with a section about "Committing with Multiple Authors", and fixed grammar and sentence structure.

monicawheeler commented on general/git.md:
```diff
@@ -0,0 +1,180 @@
+Here is a set of commit titles that are inconsistent with varying
+degrees of information.
+
+#### Not Recommended
```
👍🏻

monicawheeler commented on general/git.md:
```diff
@@ -0,0 +1,180 @@
+
+[pope-git-commit]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
+
+### Committing with Multiple Authors
```
👍🏻
